File: 05_sparse_check.py
def isSparse(matrix):
    rows, cols = len(matrix), len(matrix[0])
    zero_count = sum(row.count(0) for row in matrix)
    return zero_count > (rows * cols) // 2

mat = [[0,0,3],[0,5,0],[0,0,0]]
print("Is Sparse:", isSparse(mat))

File: 05_sparse_check.c
#include <stdio.h>

int main() {
    int m, n, count=0;
    printf("Enter rows and cols: ");
    scanf("%d %d", &m, &n);
    int arr[m][n];
    printf("Enter matrix:\n");
    for(int i=0; i<m; i++)
        for(int j=0; j<n; j++) {
            scanf("%d", &arr[i][j]);
            if(arr[i][j] == 0) count++;
        }
    if(count > (m*n)/2)
        printf("Matrix is sparse\n");
    else
        printf("Matrix is not sparse\n");
    return 0;
}